build_exercise_proj/CMakeLists.txt

```diff
@@ -1,24 +1,25 @@
-cmake_minimum_required(VERSION 3.1)
-project(build_exercise)
+cmake_minimum_required(VERSION 3.1) 
+project(build_exercise) 
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON) 
 
-set(_INC_FOLDER_NAME include)
-set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME})
-set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(_INC_FOLDER_NAME include) 
+set(_INC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${_INC_FOLDER_NAME}) 
+set(_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src) 
 
 
-add_executable(
-    ${PROJECT_NAME}     
-        ${_SRC_DIR}/main.cpp
+add_executable( 
+    ${PROJECT_NAME}      
+        ${_SRC_DIR}/main.cpp 
 )
 
-add_subdirectory(tools_lib_proj)
+add_subdirectory(tools_lib_proj) 
 
-target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj)
+target_include_directories(${PROJECT_NAME} PUBLIC tools_lib_proj) 
 
-target_link_directories(${PROJECT_NAME} PRIVATE tools_lib_proj)
+target_link_directories(${PROJECT_NAME} PRIVATE tools_lib_proj) 
 
-target_link_libraries(
-    ${PROJECT_NAME} tools 
-)
\ No newline at end of file
+target_link_libraries( 
+    ${PROJECT_NAME} tools  
+) 
```

Comment:
This should be deleted.
You already have invoked "target_link_libraries" for the project "tools_lib_proj".

Additionally, there is no "tools" project.
The project name you've used is "tools_lib_proj" (the argument provided to the "project" command)
```suggestion
```